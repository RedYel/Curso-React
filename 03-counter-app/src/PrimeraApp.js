// FC
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( { saludo, subtitulo } ) => {

    //console.log( props );
    return (
        //se elimino el Fragment        
        <>
            <h1>{ saludo }</h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
            <p>{subtitulo}</p>
        </>        
    );
    
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;