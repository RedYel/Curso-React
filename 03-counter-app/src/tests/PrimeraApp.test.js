import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import PrimeraApp from "../PrimeraApp";

describe(' Pruebas en primera App', () => {
  
    // test(' debe de mostrar el mensaje de holi ', () => {
    //    const saludo = 'Hola soy lele';

    //    const { getByText } = render( <PrimeraApp saludo={ saludo }/> )

    //    expect( getByText(saludo) ).toBeInTheDocument();
    // });
    test(' Debe de mostrar <primeraApp/> correctamente', () => {
        
        const saludo = 'Hola, soy lele';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } />);

        expect( wrapper ).toMatchSnapshot();

    });
    
    test(' Debe de mostrar ell subtitulo enviado por props', () => {
      
        const saludo = 'Hola soy lele';
        const subtitulo= 'el subtitulo'
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={subtitulo}
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        console.log( textoParrafo );

        expect(  textoParrafo ).toBe( subtitulo );
    });
    
    
});
