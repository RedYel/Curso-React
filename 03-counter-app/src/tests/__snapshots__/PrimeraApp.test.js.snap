// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en primera App  Debe de mostrar <primeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy lele
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
